### ISSUE TITLE
Retrain dataset: which patch to choose if Max-Max and Max-Min produce contradicting results for one patch

### ISSUE THREAD

**Issue by notu-ngoctrung** [ThoroughImages/CAMEL#7]
Hi!
Thank you for your previous support! I'm currently having a question about generating dataset for retrain phase, hoping to get the clarification from you.

When combining the predictions of cMIL models to have a training dataset for retrain, what should we do if the models produce contradicting prediction results for a particular patch/ instance (e.g. Max-Max predicts a patch as NC while Max-Min predicts as CA)? Let's say if these results aren't considered as confusing patches. Do we include that patch in both NC & CA class in the training set?

Thank you so much! Look forward to hearing from you

**Comment by thuxugang** [comment]
If the image label is 0, we simply discard the instance that predicted as CA. So this situation will not occur. Therefore, the contradicting instance only exists in the CA image (image label is 1). In the retrain phase, for practical usage, whether discard the NC instance in the CA image depends on different data sets (i.e. whether the false-positive problem is severe after retrained, this usually caused by data imbalance). 
Actually, we didn't notice this situation in CAMEL. I think we included the patch in both NC & CA class in the training set. However, we believe a better idea is to compare their softmax probabilities.
